templates/AMI-Template-Instances.cfn.yml.j2

```diff
@@ -48,19 +82,29 @@ Resources:
           Ebs:
             Encrypted: true
             DeleteOnTermination: true
-            VolumeSize: {{ disk_size }}
+            VolumeSize: !Ref {{ layer }}EC2SecondaryDiskSize
             VolumeType: "gp2"
       IamInstanceProfile:
-        Fn::ImportValue: !Sub "${Environment}${ApplicationName}{{ instance_profile_prefix | default('') }}InstanceProfileName"
-      # Amazon Linux 2023: al2023-ami-2023.4.20240611.0-kernel-6.1-x86_64
+        Fn::ImportValue:
+          Fn::Join:
+            - ""
+            - - !Ref Environment
+              - !Ref ApplicationName
+              - !Ref {{ layer }}InstanceProfileName
+      # Amazon Linux 2023: al2023-ami-2023.6.20250317.2-kernel-6.1-x86_64
       ImageId: "{{ base_ami }}"
       InstanceType: "t3a.small"
       KeyName: !Ref DefaultEC2Key
       NetworkInterfaces:
         - DeviceIndex: 0
           GroupSet:
             - Fn::ImportValue: !Sub "${Environment}CKANManagementSG"
-            - Fn::ImportValue: !Sub "${Environment}CKAN{{ security_group }}"
+            - Fn::ImportValue:
```

Comment:
sub is nicer than import for reading and searching

Reply:
Agreed, but see my other comment.